libraries/matrix/api/src/main/kotlin/io/element/android/libraries/matrix/api/room/DmChecker.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2024 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.libraries.matrix.api.room
+
+/**
+ * Verifies if a room is considered a direct message one for Element.
+ */
+object DmChecker {
```

Comment:
I would not introduce a class for this, but rather create some extensions, like `RoomInfo.isElementDm()`, `MatrixRoomInfo.isElementDm()`, etc.

If all those extensions have the same name, and all in the same file, I think it can work.

Reply:
I'm afraid if we have 3, 4 of these extensions with exactly the same code inside them we might want to update them and forget about some of them. That was why I created this object.

We could make it so the extensions all called the same private function underneath though, so we could ensure the check is done with the same piece of code for every one of them.

Reply:
. o O (Why do we have so many types to describe the same thing?)

Still it will be more clean and less error prone to use an extension at the call site.

Reply:
There are a couple of places where it wouldn't be possible to just use extensions:

- `RoomSummaryDetailsFactory.create`: it uses a Rust `RoomInfo` which is not available in the `matrix.api` module.
- `NotificationMapper.map`, we only have the `NotificationRoomInfo`, which also comes from Rust and is not available in `matrix.api`.

Either we keep the `DmChecker` object for those or we create a public `fun isRoomDm(isDirect: Boolean, activeMemeberCount: Int): Boolean` that can be called for other cases.

WDYT?